packages/terra-application-header-layout/src/ApplicationHeaderLayout.jsx

```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import AppDelegate from 'terra-app-delegate';
+
+import styles from './ApplicationHeaderLayout.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * The AppDelegate instance provided by the containing component. If present, its properties will propagate to the children components.
+   * */
+  app: AppDelegate.propType,
```

Comment:
It still feels weird to me to expose the app and layoutConfig props to what is a pure UI component, not a framework one. This prop forwarding would be a task better suited for a parent component. 

My main concerns are the development of future templates (this logic would be kind of suck to duplicate everywhere), flexibility of use (consumers cannot control this prop flow, and if they need to for some reason, they have to roll a new header layout entirely), and the potential for use outside of the framework we're developing.

Reply:
So you're suggesting that we'd create a HOC wrapper of this with the responsibility of simply taking the same props and adding the app and config props to those children? What would you call this thing and where would you suggest it lives?

Reply:
We talked about this offline, but for posterity:

We're going to have an ApplicationHeader or something that is going to consume the ApplicationHeaderLayout and determine what components are going into various layout regions. This ApplicationHeader component is what will be dropped into the Layout and what will be receiving `app` and `layoutConfig` props.

My point is that this ApplicationHeader is going to be instantiating the various logo, navigation, and utility components that are to be placed in the layout. Why shouldn't it just pass along the `layoutConfig` and `app` props to those pieces at instantiation and save the ApplicationHeaderLayout the trouble? It allows the flow of information to be controlled at a higher level and would allow subsequent templates to be lighter (if they are ever developed).

At the same time, if we don't think the ApplicationHeader will ever use a different layout, or if we don't think the ApplicationHeaderLayout will be used outside of the ApplicationHeader, I don't have a problem with this staying as is. I would just expect the ApplicationHeader and the ApplicationHeaderLayout to live in the same package.